soweego/linker/classify.py

```diff
@@ -67,50 +76,112 @@ def _upload(predictions, catalog, sandbox):
 
 
 def execute(catalog, entity, model, name_rule, threshold, dir_io):
-    wd_reader = workflow.build_wikidata(
-        'classification', catalog, entity, dir_io)
-    wd_generator = workflow.preprocess_wikidata('classification', wd_reader)
+    complete_fv_path = os.path.join(dir_io, constants.COMPLETE_FEATURE_VECTORS %
+                                    (catalog, entity, 'classification'))
+    complete_wd_path = os.path.join(dir_io, constants.COMPLETE_WIKIDATA_CHUNKS %
+                                    (catalog, entity, 'classification'))
+    complete_target_path = os.path.join(dir_io, constants.COMPLETE_TARGET_CHUNKS %
+                                        (catalog, entity, 'classification'))
 
     classifier = joblib.load(model)
     rl.set_option(*constants.CLASSIFICATION_RETURN_SERIES)
-    for i, wd_chunk in enumerate(wd_generator, 1):
-        # TODO Also consider blocking on URLs
 
-        samples = blocking.full_text_query_block(
-            'classification', catalog, wd_chunk[constants.NAME_TOKENS],
-            i, target_database.get_entity(catalog, entity), dir_io)
+    # check if files exists for these paths. If yes then just
+    # preprocess them in chunks instead of recomputing
+    if all(os.path.isfile(p) for p in [complete_fv_path,
+                                       complete_wd_path,
+                                       complete_target_path]):
+
+        LOGGER.info(
+            'Using previously cached version of the classification dataset')
+
+        fvectors = pickle.load(open(complete_fv_path, 'rb'))
```

Comment:
I would use `pandas.read_pickle` here and in the following calls.

Reply:
Thanks for the suggestion @marfox ! 
Changes have been implemented b431993 and ae5e215